docs/site/Deploying_to_ibm_cloud_kubernetes.md

```diff
@@ -0,0 +1,707 @@
+---
+lang: en
+title: 'Deploying to IBM Cloud on Kubernetes'
+keywords: LoopBack 4.0, LoopBack 4, Kubernetes, IBM Cloud, Docker, Container Registry
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/deploying_to_ibm_cloud_kubernetes.html
+---
+
+## Overview
+
+This tutorial shows how to deploy a [LoopBack4](http://v4.loopback.io/) application onto [Kubernetes](http://kubernetes.io/) on the [IBM Cloud](http://cloud.ibm.com).
+
+## Prerequisite
+
+You’ll need the following:
+
+- Node.js 8.9 or higher
+- [Docker 18.06 or higher](https://docs.docker.com/install/)
+- [IBM Cloud CLI, Container registry CLI, etc](https://cloud.ibm.com/docs/cli/index.html#overview)
+- [Kubernetes CLI (kubectl)](https://kubernetes.io/docs/reference/kubectl/overview/)
+- LoopBack 4 CLI
+
+Let's install the LoopBack 4 CLI :
+
+```
+npm i -g @loopback/cli
+```
+
+After you have scaffolded a LoopBack4 application, you’ll get a default endpoint /ping. For simplicity sake, we’ll use this.
+
+## Step 1: Scaffold LoopBack 4 app
+
+Run ````lb4 app```` command.
+
+````
+$ lb4 app
+? Project name: deploy-test
+? Project description: loopback example
+? Project root directory: deploy-test
+? Application class name: DeployTestApplication
+? Select features to enable in the project Enable tslint, Enable prettier, Enable mocha, Enable loopback
+Build, Enable vscode, Enable repositories, Enable services
+...
+Application deploy-test was created in deploy-test.
+Next steps:
+$ cd deploy-test
+$ npm start
+````
+
+Navigate to the main directory of your project
+
+````
+cd deploy-test
+````
+
+Let's **not** run ````npm start```` yet, we have some changes to make to certain files.
+
+## Step 2: Modify deploy-test/index.js
+
+It is important for the host to be ````'0.0.0.0'```` and the port to be consistent in several places. We will use a port of ````8080````.
+
+````
+  const config = {
+    rest: {
+      port: 8080,
+      host: '0.0.0.0',
+      openApiSpec: {
+        // useful when used with OASGraph to locate your application
+        setServersFromRequest: true,
+      },
+    },
+  };
+````
+
+## Step 3: Modify deploy-test/src/index.ts
+
+It is important for the host to be ````'0.0.0.0'```` and the port to be consistent in several places. We will use a port of ````8080````.
+
+
+````
+import {DeployTestApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+export {DeployTestApplication};
+
+export async function main(options: ApplicationConfig = {}) {
+
+  // --------- ADD THIS SNIPPET -------------
+if (!options) options = {};
+if (!options.rest) options.rest = {};
+options.rest.port = 8080;
+options.rest.host = "0.0.0.0";
+// ----------------------------------------
```

Comment:
We have configured options in `index.js` as part of step 2. Is step 3 still needed?

Reply:
not sure. It worked when I edited both. I will try next week.

Reply:
did you find time to try ?

Reply:
I've finally had the chance to experiment with this a bit more. It turns out @raymondfeng  is correct.  "Modify deploy-test/src/index.ts" is not needed if I have done "Modify deploy-test/index.js".

Reply:
@JesusTheHun ^^